src/go/k8s/pkg/resources/service.go

```diff
@@ -108,3 +109,14 @@ func (r *ServiceResource) Kind() string {
 	var svc corev1.Service
 	return svc.Kind
 }
+
+// HeadlessServiceFQDN returns fully qualified domain name for headless service.
+// It can be used to communicate between namespaces if the network policy
+// allows it.
+func (r ServiceResource) HeadlessServiceFQDN() string {
+	// TODO Retrieve cluster domain dynamically and remove hardcoded cluster.local
```

Comment:
i bet we could detect if we are running within amazon (EKS) or google (GKE) and they have a way to retrieve this information too and the dfeault would be this which is cool. 

Just a thought. nothing to do.